test/upgrade/prober/wathola/event/services.go

```diff
@@ -91,7 +97,7 @@ func (f *finishedStore) RegisterFinished(finished *Finished) {
 	time.Sleep(d)
 	receivedEvents := f.steps.Count()
 	if receivedEvents != finished.Count {
-		f.errors.throw("expecting to have %v unique events received, "+
+		f.errors.throw(TagMissing, "expecting to have %v unique events received, "+
```

Comment:
Here you put a Missing tag, but this error is there if there are less or more events than expected. Receiving more events than expected is also a possibility, and one which we should report as error.

My general comment describes the proposition on creating here an error dedicated to the situation of receiving unexpected events. We already report each missing event separately, so it usually doubles that information.

Reply:
Make sense. I think it skipped my mind when I worked on this because I had never really seen unexpected events in our tests. Will check different case.

Reply:
Yeah, that why we are not using it in the wild. I plan to run that prober also on clusters with some real usage, then, maybe we could see it. Hopefully not, though.

Reply:
That would sounds a bit bug.

Reply:
Fixed as suggested.

Reply:
Use `unexpected` as suggested.